app/v2/controllers/V2DeparturesController.scala

```diff
@@ -241,9 +243,37 @@ class V2DeparturesControllerImpl @Inject() (
           )
     }
 
-  def attachMessage(departureId: DepartureId): Action[Source[ByteString, _]] = Action(streamFromMemory) {
-    implicit request =>
-      request.body.runWith(Sink.ignore)
-      Accepted(Json.obj("version" -> 2))
-  }
+  def attachMessage(departureId: DepartureId): Action[Source[ByteString, _]] =
+    (authActionNewEnrolmentOnly andThen messageSizeAction()).async(streamFromMemory) {
+      implicit request =>
+        withTemporaryFile {
+          (temporaryFile, source) =>
+            implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+            (for {
+              messageType <- xmlParsingService.validateMessageType(source).asPresentation
+              _           <- validationService.validateXml(messageType, source).asPresentation
+
+              fileSource = FileIO.fromPath(temporaryFile)
```

Comment:
This isn't going to work -- `source` can only be used once because it's a request stream.

I was originally not going to merge https://github.com/hmrc/common-transit-convention-traders/pull/299 until this was done because I didn't want to give you too much of a moving target, but it's clear that actually it'd be better if I did. Once I have merged that (it'll be soon), you'll need to update this to use that -- effectively it'll remove the confusing use of multiple sources and you won't need to worry about it.

Reply:
done the changes